### PR TITLE
feat: lint option for javascript template

### PR DESCRIPTION
**Pull request by Mazuh**
resolves https://github.com/fastify/fastify-cli/issues/363

#### Checklist

- [x] run `npm run unit`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

for typescript, see: https://github.com/fastify/fastify-cli/pull/512

### CODE REVIEW

File: test/generate.test.js [hunk 1]
```diff
@@ -159,6 +171,23 @@ function define (t) {
     })
   }
 
+  function verifyStandardlintInPkg (t) {
+    return new Promise((resolve, reject) => {
+      const pkgFile = path.join(workdir, 'package.json')
+
+      readFile(pkgFile, function (err, data) {
```
Comment (mcollina) [inline test/generate.test.js:178]: Could you please use `fs.promises.readFile()`.

**Review by mcollina — CHANGES_REQUESTED** [review]

File: test/generate.test.js (hunk 1 above)
Comment (Mazuh) [reply test/generate.test.js:178]: sure, I just tried to follow the existing code pattern, but it's fixed with what you asked now.

**Review by mcollina — APPROVED** [review]
lgtm

**Review by climba03003 — CHANGES_REQUESTED** [review]
I would change the option to `lint` which can be extendable in future.

**Comment by climba03003** [comment]
> IMHO people that need something beyond the standard linter will probably do that with their own custom *rc files (.prettierc, .eslintrc), each one requiring extra dependencies (every eslint project has tons of different plugins listed on devdependencies), at least that's my experience, not to mention that eslint/prettier CLIs are already very customizable by themselves.

It is a `string` option which mean it can be extended like `--lint=standard`, `--lint=eslint`, etc.

File: args.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ module.exports = function parseArgs (args) {
       'populate--': true
     },
     number: ['port', 'inspect-port', 'body-limit', 'plugin-timeout'],
-    string: ['log-level', 'address', 'socket', 'prefix', 'ignore-watch', 'logging-module', 'debug-host', 'lang', 'require'],
+    string: ['log-level', 'address', 'socket', 'prefix', 'ignore-watch', 'logging-module', 'debug-host', 'lang', 'require', 'standardlint'],
```
Comment (climba03003) [inline args.js:15]: At least I see you declare it as string option.

Comment (Mazuh) [reply args.js:15]: Good point. That's because I started to code thinking of it to be a string indeed, then I thought "if someone really tries to put more options here, it will be annoying AF to decide and maintain each non-standard option" and so I changed my mind.

but, again, I'm just waiting for some agreed final design choice from all reviewers, it should be an easy code change anyway.

File: generate.js [hunk 3]
```diff
@@ -149,7 +153,25 @@ function cli (args) {
...
+  if (opts.lang === 'ts' || opts.lang === 'typescript') {
+    template = typescriptTemplate
+  } else {
+    template = { ...javascriptTemplate }
+
+    if (opts.standardlint) {
+      template.scripts = {
+        ...template.scripts,
+        pretest: 'standard',
+        lint: 'standard --fix'
```
Comment (Eomm) [inline generate.js:166]: In this org we use:

```
lint -> run the linter
lint:fix -> run the linter's fixes
```

**Review by Eomm — COMMENTED** [review]
> tldr: I tried to avoid a lot of headache for maintainers.

**+1**

I must say that it would be awesome to think of a "pluggable" cli (or a yeomen generator? 🤔 )

**Review by mcollina — APPROVED** [review]
lgtm

File: generate.js (hunk 3 above)
Comment (Mazuh) [reply generate.js:166]: @Eomm, I prefer such pattern too, but this is for the template, not the org project itself, and for the typescript template it's already just `lint` instead of `lint:fix`, and just `lint` is already what's recommended by the readme right now.

so here I ask two questions:
- should I proceed changing it to `lint:fix`? 
- if yes, then should I update typescript template too (along with its docs) (and readme suggestion)?

**Comment by climba03003** [comment]
> I must say that it would be awesome to think of a "pluggable" cli (or a yeomen generator? 🤔 )

@Eomm I recommend to check this PR https://github.com/fastify/fastify-cli/pull/491

**Comment by Mazuh** [comment]
@climba03003 @Eomm any feedback on this? should I assume that this PR should be close (it looks like the issue is being worked on in another PR or something)?

File: generate.js (hunk 3 above)
Comment (Eomm) [reply generate.js:166]: I reread the readme and I would not change it as it should be already discussed in past

**Review by Eomm — APPROVED** [review]

**Review by Eomm — APPROVED** [review]
LGTM when ci green

**Comment by simoneb** [comment]
@Mazuh let's figure out the windows problem first

**Comment by Mazuh** [comment]
it was indeed just a testing issue, the `rimraf.sync(...)` that runs before each test behaves a little bit differently on windows, all others test cases were running fine but my little one stepped on a race condition and I was trying to generate the project in a busy directory.

now I have my ubuntu and my windows with `npm t` returning 0 (ok), so the same should happen with the pipeline, I'm going back to my macbook to confirm that everything is ok there too.

Captura de tela 2022-06-21 081851

**Comment by mcollina** [comment]
@simoneb you good to land this?

File: help/generate.txt [hunk 4]
```diff
@@ -11,3 +11,6 @@ OPTS
 
   --lang=ts, --lang=typescript
       use the TypeScript template
+
+  --standardlint
```
[the remark this answers is not in the document]
Comment (simoneb) [reply help/generate.txt:15]: also, any reason why we don't do it for the ts template?

[the remark this answers is not in the document]
Comment (Mazuh) [reply help/generate.txt:15]: yes, tests will pass but it won't work so easily like js support, I opened another PR for further discussions: https://github.com/fastify/fastify-cli/pull/512

[the remark this answers is not in the document]
Comment (Mazuh) [reply help/generate.txt:15]: and about the option name, it's a consequence of a few discussions above and the conclusion is to **narrow** this feature to a simple yes-or-no support, while some other future feature (afaik, that already has a PR in WIP) will add a more broad support to other linter options.

**Review by simoneb — APPROVED** [review]

**Comment by simoneb** [comment]
@mcollina yes LGTM
